daemon/loop.go

```diff
@@ -149,6 +150,52 @@ func (d *Daemon) pullAndSync(logger log.Logger) {
 		}); err != nil {
 			logger.Log("err", err)
 		}
+
+		// Find notes in revisions.
+		for i, rev := range revisions {
+			// Skip last revision (since we requested HEAD~1 above)
+			if i == len(revisions)-1 {
```

Comment:
See this line: https://github.com/weaveworks/flux/blob/a532aac4e14410abfb1d7c1186000ff601a4b599/daemon/loop.go#L97
It requests two revisions, HEAD and HEAD~1. We need to skip HEAD~1 to avoid duplicates.